### ISSUE TITLE
Support WITH templates

### ISSUE THREAD

**Issue by hagen1778** [VictoriaMetrics/grafana-datasource#35]
[WITH templates](https://play.victoriametrics.com/promql/expand-with-exprs?query=WITH+%28%0D%0A++++maxRate+%3D+1e9+%2F+8%2C+%23+Gigabit+network%0D%0A++++commonFilters+%3D+%7Binstance%3D%7E%22%24node%3A%24port%22%2Cjob%3D%7E%22%24job%22%7D%2C%0D%0A++++networkUtilization%28bytesTotal%2C+maxRate%29+%3D+ru%28maxRate+-+rate%28bytesTotal%7BcommonFilters%7D%5B5m%5D%29%2C+maxRate%29%0D%0A%29+networkUtilization%28node_network_receive_bytes_total%2C+maxRate%29&commit=Expand) in VictoriaMetrics help to simplify complex queries. Would be nice to introduce an option to define global or per-panel expressions, so users can define WITH templates once and then re-use them when needed.

**Comment by hagen1778** [comment]
Thanks for https://github.com/VictoriaMetrics/grafana-datasource/pull/72 @Loori-R!
I think it is already great, but we should polish it a bit.

- [ ] Support comments for WITH template expressions. For example:
```
# cf filters by job and instance
cf = {job=~"$job", instance=~"$instance"},
# rateBy calculates rate and groups by metric name and group label. Use it instead of rate.
rateBy(metric, group) = sum(rate(metric{cf})) by(group)
```
When using these expressions in query panel, it would be nice to display the `# comments` in the popup windows with autosuggestion. So user could read the comment and get more idea about this custom function or filter

- [ ] Allow auto-complete inside curly brackets for filters defined in WITH expressions. In the example above, we defined `cf` filter. It would be great for plugin to suggest it along with other labels when user opens a curly braces `{` in query.

- [ ] Add docs explaining how to use WITH templates integration. The doc should cover how to define WITH expressions, how to work with comments, how to validate, and how to use defined expressions inside panel queries. Such docs would significantly improve users experience and understanding of this feature.

**Comment by hagen1778** [comment]
The issue was addressed in https://github.com/VictoriaMetrics/grafana-datasource/pull/85 and improvements will be included into the next plugin release.

**Comment by Loori-R** [comment]
This feature is available starting from [v0.3.0](https://github.com/VictoriaMetrics/grafana-datasource/releases). Closing the feature request as done.

**Comment by Karsonito** [comment]
Hello!
Grafana 9.5.7
VM DS 0.3.0
![image](https://github.com/VictoriaMetrics/grafana-datasource/assets/2379481/52e487c1-7a09-4985-8d8c-1580f22cbc3d)

This is my first experience with VM DS. May be doing something wrong :)
